iceoryx2-ffi/cxx/src/error_string.cpp

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+#include "iox2/config_creation_error.hpp"
+#include "iox2/connection_failure.hpp"
+#include "iox2/enum_translation.hpp"
+#include "iox2/iceoryx2.h"
+#include "iox2/listener_error.hpp"
+#include "iox2/node_failure_enums.hpp"
+#include "iox2/node_wait_failure.hpp"
+#include "iox2/notifier_error.hpp"
+#include "iox2/publisher_error.hpp"
+#include "iox2/semantic_string.hpp"
+#include "iox2/service_builder_event_error.hpp"
+#include "iox2/service_builder_publish_subscribe_error.hpp"
+#include "iox2/service_error_enums.hpp"
+#include "iox2/subscriber_error.hpp"
+#include "iox2/waitset_enums.hpp"
+
+namespace iox2 {
+
+auto error_string(const iox2::ConfigCreationError& error) -> const char* {
```

Comment:
All those implementation should be changed to the format:

```cxx
template <>
constexpr auto from<iox2::ConfigCreationError, const char *>(const iox2::ConfigCreationError value) noexcept -> const char * {
  return iox2_config_creation_error_string(iox::into<iox2_config_creation_error_e>(value));
}
```

Reply:
Could you also rename the file into `enum_string.cpp`. It makes sense to have this conversion also for non error enums like `ServiceType` for instance.

Reply:
@elfenpiff Could the generalization to all enum types be left for a separate PR? If I change this I would need to add the implementation for getting the string representation of all enums as well.

Reply:
@elBoberido you need to solve the tie.

I would prefer `iox::into<const char*>`. For me, it is clear that the enum should be converted into a human-readable string.

Reply:
@orecham @elfenpiff well, I would consider both approaches bad. What I want is `IOX_LOG(DEBUG, "Error code: " << result.error())`. It's a pity we re-invent the wheel with the logger, although we already have something in hoofs, which is more ergonomic than what we have now in iceoryx2.

Reply:
@elBoberido This would be nice, but users not using the iceoryx logger would run into trouble.

Reply:
Yeah, for them there would need to be a way to get the string.

I still do not have a clear favorite. Both approaches have pros and cons but I'm leaning towards the `into` approach
- we are already using it in our code base, so the users should be familiar with the concept
- we can use the same concept for conversion into `iox::string`, `std::string` or anything else if it is not possible to have a static conversion and it's clear to the user that a conversion has some cost

Reply:
I agree with @orecham that this

```cxx
iox::into<const char*>(result.error()))
```

is a bit more cumbersome to use than this

```cxx
error_string(result.error())
```

But the `iox::into` is the iceoryx idiomatic way of casting things so therefore I would stick with it - this would be also the first thing I would try if I wanted to cast X to a string. In this case, I prefer consistency over ergonomic handling.

Reply:
@elfenpiff @elBoberido Changed to use `iox::into`.